x-pack/plugins/index_lifecycle_management/public/application/services/api.ts

```diff
@@ -4,68 +4,77 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+import { UiStatsMetricType } from '@kbn/analytics';
+
 import {
   UIM_POLICY_DELETE,
   UIM_POLICY_ATTACH_INDEX,
   UIM_POLICY_ATTACH_INDEX_TEMPLATE,
   UIM_POLICY_DETACH_INDEX,
   UIM_INDEX_RETRY_STEP,
 } from '../constants';
-
 import { trackUiMetric } from './ui_metric';
-import { sendGet, sendPost, sendDelete } from './http';
+import { sendGet, sendPost, sendDelete, useRequest } from './http';
 
 export async function loadNodes() {
   return await sendGet(`nodes/list`);
 }
 
-export async function loadNodeDetails(selectedNodeAttrs) {
+export async function loadNodeDetails(selectedNodeAttrs: string) {
   return await sendGet(`nodes/${selectedNodeAttrs}/details`);
 }
 
 export async function loadIndexTemplates() {
   return await sendGet(`templates`);
 }
 
-export async function loadPolicies(withIndices) {
+export async function loadPolicies(withIndices: boolean) {
   return await sendGet('policies', { withIndices });
 }
 
-export async function savePolicy(policy) {
+export async function savePolicy(policy: any) {
   return await sendPost(`policies`, policy);
 }
 
-export async function deletePolicy(policyName) {
+export async function deletePolicy(policyName: string) {
   const response = await sendDelete(`policies/${encodeURIComponent(policyName)}`);
   // Only track successful actions.
-  trackUiMetric('count', UIM_POLICY_DELETE);
+  trackUiMetric('count' as UiStatsMetricType, UIM_POLICY_DELETE);
```

Comment:
nit; since these are all `'count'` perhaps we can create a shared variable `const count: UiStatsMetricType = 'count';` and reuse that. This way we can remove the `as` declarations.

Reply:
Yes, I agree that using `as` is not the best way. I actually saw CJ using an enum in this same file TS conversion in #70382.